README.rst

```diff
@@ -0,0 +1,325 @@
+============================================
+Batch Data Processing in CDAP using Workflow
+============================================
+
+The Workflow system in `Cask Data Application Platform (CDAP) <http://cdap.io>`__
+allows specifying, executing, scheduling, and monitoring complex series of jobs
+and tasks. In this guide, you will learn how it can facilitate the execution of
+`MapReduce <http://research.google.com/archive/mapreduce.html>`__
+programs in parallel based on the evaluation of the boolean predicate.
+
+What You Will Build
+===================
+
+This guide will take you through building a
+`CDAP application <http://docs.cdap.io/cdap/current/en/developers-manual/building-blocks/applications.html>`__
+that uses ingested raw purchase events (of the form '<name> bought <n> <items> for $<price>') to compute
+the total purchases made by user along with the total purchases made for the particular product.
+
+You will:
+
+- Build ``PurchaseEventParser``, a
+  `MapReduce program <http://docs.cdap.io/cdap/current/en/developers-manual/building-blocks/mapreduce-programs.html>`__
+  to parse the raw purchase events and create ``Purchase`` objects for them;
+- Build ``PurchaseCounterByUser``, a
+  `MapReduce program <http://docs.cdap.io/cdap/current/en/developers-manual/building-blocks/mapreduce-programs.html>`__
+  to count the purchases made per user;
+- Build ``PurchaseCounterByProduct``, a
+  `MapReduce program <http://docs.cdap.io/cdap/current/en/developers-manual/building-blocks/mapreduce-programs.html>`__
+  to count the purchases made per product;
+- Build ``PurchaseWorkflow``, a
+  `Workflow program <http://docs.cdap.io/cdap/current/en/developers-manual/building-blocks/workflows.html>`__
+  which will first execute the MapReduce program ``PurchaseEventParser``. Then if the predicate ``PurchaseEventVerifier``
```

Comment:
Then, if the predicate...evaluates to true, the workflow will in parallel execute the MapReduce programs...otherwise, it will execute the action...